packages/grid/x-data-grid/src/components/base/GridOverlays.tsx

```diff
@@ -29,24 +38,23 @@ function GridOverlayWrapper(props: React.PropsWithChildren<{}>) {
 
   let height: React.CSSProperties['height'] = viewportInnerSize?.height ?? 0;
   if (rootProps.autoHeight && height === 0) {
-    height = 'auto';
+    height = getMinimalContentHeight(apiRef); // Give room to show the overlay when there no rows.
```

Comment:
The trick with `position: absolute; top: 0; bottom: 0; height: auto` to get full height won't work anymore, hence the change.